src/pages/transactions/[tid].tsx

```diff
@@ -0,0 +1,698 @@
+            tokenPrice={data.transactions.tokenPrice}
+            valuePrice={data.transactions.valuePrice}
+            status={data.transactions.status}
+            blockNumber={data.number}
+            timestamp={data.timestamp}
+            transactionFee={data.transactions.transactionFee}
+            block={data.transactions.block}
+            transactionType={data.transactions.transactionType}
+            nonce={data.transactions.nonce}
+            address={data.transactions.address}
+            type={data.type}
+          />
+          <div className="border-[1px] border-black-600 lg:my-11 md:mt-9 md:mb-11 mt-10 mb-[52px]" />
+          <TransactionDetailSegmentTwo
+            gasPrice={data.transactions.gasPrice}
+            gasLimit={data.gasLimit}
+            gasUsed={data.gasUsed}
+            gasUsedPercentage={data.gasUsedPercentage}
+            from={data.transactions.address.from}
+            to={data.transactions.address.to}
+            forToken={data.forToken}
+            type={data.type}
+            hex={data.hex}
+          />
+        </div>
+      </GradientCardContainer>
+    </Container>
+  );
+}
+
+interface TransactionDetailSegmentOneProp {
+  transactionId: string;
+  tokenPrice: {
+    value: string;
+    symbol: string;
+  };
+  valuePrice: string;
+  status: string;
+  blockNumber: string;
+  timestamp: string;
+  transactionFee: {
+    value: string;
+    symbol: string;
+  };
+  block: string;
+  transactionType: string;
+  nonce: string;
+  address: {
+    from: string;
+    to: string;
+    contractAddress: string;
+  };
+  type: string;
+}
+
+function TransactionDetailSegmentOne({
+  transactionId,
+  tokenPrice,
+  valuePrice,
+  status,
+  blockNumber,
+  timestamp,
+  transactionFee,
+  block,
+  transactionType,
+  nonce,
+  address,
+  type,
+}: TransactionDetailSegmentOneProp) {
+  const [isFromAddressCopied, setIsFromAddressCopied] = useState(false);
+  const [isToAddressCopied, setIsToAddressCopied] = useState(false);
+  const [isTransactionIdCopied, setIsTransationIdCopied] = useState(false);
+
+  return (
+    <div className="flex flex-col gap-y-10">
+      {/* first row */}
+      <div className="flex flex-col lg:flex-row md:flex-row gap-y-[41.5px]">
+        {/* 1st flex */}
+        <div className="flex flex-col grow gap-y-2">
+          {isTransactionIdCopied ? (
+            <div className="flex flex-row gap-x-2.5 items-center">
+              <LinkText
+                testId="transaction-id-copied"
+                label={fixedTitle.copied}
+                href={`/address/${transactionId}`}
+              />
+              <GreenTickIcon data-testid="transaction-id-copied-green-tick-icon" />
+            </div>
+          ) : (
+            <div className="flex flex-row gap-x-2.5 items-center">
+              <LinkText
+                testId="trasaction-id"
+                label={truncateTextFromMiddle(transactionId)}
```

Comment:
in design, instead of last and first 5 places we have to show first and last 11 place, can we add second parameter as 11 in `truncateTextFromMiddle` func

Reply:
added useWindowDimension hook to get screen size and truncate accordingly based on screen size